File: java_solutions/168.excel-sheet-column-title.java
/*
 * @lc app=leetcode id=168 lang=java
 *
 * [168] Excel Sheet Column Title
 */

class Solution {
    public String convertToTitle(int col) {
        return Stream.iterate(col, c -> c > 0, c -> (c - 1) / 26) .map(c -> String.valueOf((char) ('A' + (c - 1) % 26))) .collect(Collectors.collectingAndThen( Collectors.joining(), sb -> new StringBuilder(sb).reverse().toString()));
    }
}

File: kotlin_solutions/168.excel-sheet-column-title.kt
/*
 * @lc app=leetcode id=168 lang=kotlin
 *
 * [168] Excel Sheet Column Title
 */

class Solution { fun convertToTitle(columnNumber: Int) = generateSequence(columnNumber - 1) { (it / 26) - 1 } .takeWhile { it >= 0 } .distinct() .fold("") { acc, i -> acc + ('A' + i % 26) } .reversed() }
